src/platform_impl/linux/mod.rs

```diff
@@ -606,45 +606,51 @@ impl<T: 'static> Clone for EventLoopProxy<T> {
 }
 
 impl<T: 'static> EventLoop<T> {
-    pub(crate) fn new(attributes: &PlatformSpecificEventLoopAttributes) -> Self {
+    pub(crate) fn new(attributes: &PlatformSpecificEventLoopAttributes) -> Result<Self, String> {
         if !attributes.any_thread && !is_main_thread() {
-            panic!(
+            return Err(
```

Comment:
This is a programmer error, and is not meant to be recoverable!

Reply:
I have changed this